firmware/projects/DemoProjectRTOS/platforms/stm32f767/os.cc

```diff
@@ -0,0 +1,49 @@
+/// @author Matteo Tullo
+/// @date 2024-02-01
+
+#include "cmsis_os2.h"
+
+#include "mcal/stm32f767/os/semaphore.h"
+#include "mcal/stm32f767/os/mutex.h"
+#include "mcal/stm32f767/os/fifo.h"
+#include "mcal/stm32f767/os/timer.h"
+#include "mcal/stm32f767/os/tick.h"
+
+extern "C" {
+/**
+ * This requires extern since it is not declared in a header, only defined
+ * in cubemx/../freertos.c
+ */
+void MX_FREERTOS_Init(void);
+extern osSemaphoreId_t testSemaphoreHandle;
+extern osSemaphoreId_t testBinarySemaphoreHandle;
+extern osMutexId_t testMutexHandle;
+extern osMessageQueueId_t myTestQueueHandle;
+extern osTimerId_t messageTimerHandle;
+}
+
+namespace mcal {
+os::Semaphore sem_test{&testBinarySemaphoreHandle};
+os::Mutex mutex_test{&testMutexHandle};
+os::Fifo message_queue_test{&myTestQueueHandle};
+os::Timer timer_test{&messageTimerHandle};
+}   // namespace mcal
+
+namespace os {
```

Comment:
should rename to `bindings_os` to avoid name collisions with `mcal::os` and `shared::os`

Reply:
changed mind, leave as is